### ISSUE TITLE
ICD_LOADER_MAGIC check fails when using MoltenVK through the vulkan-loader in debug

### ISSUE THREAD

**Issue by ivovandongen** [KhronosGroup/MoltenVK#689]
```
Assertion failed: (valid_loader_magic_value(obj) && "Incompatible ICD, first dword must be initialized to " "ICD_LOADER_MAGIC. See loader/README.md for details."), function loader_init_dispatch, file /Users/ivo/git/ivovandongen/vulkan-playground/deps/vulkan-loader/loader/loader.h, line 410.
```

This occurs when using MoltenVK as an ICS through the vulkan-loader. After acquiring and subsequently freeing a commandbuffer, the object is pooled, but seems to loose the proper ICD loader magic. When acquiring another command buffer on the same command pool, the debug assert fails on https://github.com/KhronosGroup/Vulkan-Headers/blob/master/include/vulkan/vk_icd.h#L73-L76

If turning off object pooling for `CommandBuffer`s, there is no issue.

Simple reproduction (after all the setup for device through to command pool):

```
VkCommandBufferAllocateInfo allocInfo = {};
allocInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
allocInfo.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
allocInfo.commandPool = commandPool;
allocInfo.commandBufferCount = 1;

VkCommandBuffer commandBuffer;
vkAllocateCommandBuffers(device, &allocInfo, &commandBuffer);
vkFreeCommandBuffers(device, commandPool, 1, &commandBuffer);

VkCommandBuffer commandBuffer1;
vkAllocateCommandBuffers(device, &allocInfo, &commandBuffer1);
vkFreeCommandBuffers(device, commandPool, 1, &commandBuffer1);
```

Using MoltenVK 1.0.36 and Vulkan headers/loader 1.1.114.

**Comment by billhollings** [comment]
MoltenVK has never manipulated the magic number beyond its initial setting when creating new dispatchable Vulkan handles. Based on testing...the magic number seems to be coming back corrupted on objects freed by the loader.

Anyway...I think I've fixed this in PR #690 by re-establishing the magic number every time when MoltenVK returns a handle to the loader, whether the object was newly created, or retrieved from a pool.

**Comment by ivovandongen** [comment]
@billhollings Thanks for the quick fix! Working perfectly.
